### PR TITLE
Implement sort() method

### PR DESCRIPTION
**Pull request by colinodell**
Implements the sort() method in #51

### CODE REVIEW

**Comment by colinodell** [comment]
@lstrojny I took a stab at creating a sort method.  Let me know if you'd like to see any changes to this.

**Comment by colinodell** [comment]
Oh, and I should mention that I only implemented the userland function as I'm not a C developer.

File: src/Functional/Sort.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    if ($collection instanceof \Traversable) {
+        $array = iterator_to_array($collection);
+    } else {
+        $array = $collection;
+    }
+
+    $fn = $preserveKeys ? 'uasort' : 'usort';
+
+    $fn($array, function ($v1, $v2) use ($callback, $collection) {
```
Comment (lstrojny) [inline src/Functional/Sort.php]: $v1 => $left, $v2 => $left

**Comment by colinodell** [comment]
The variables have been renamed.

File: README.md [hunk 2]
```diff
@@ -376,6 +376,33 @@ F\with($value, function($value) {
...
 
+### Functional\sort()
+Sorts a collection with a user-defined function, optionally preserving array keys
+
+```php
+<?php
+use Functional as F;
+
+// Sorts a collection alphabetically
+F\sort($collection, function($value1, $value2) {
```
Comment (lstrojny) [inline README.md]: Here as well, let’s use `$left`, `$right`

File: README.md [hunk 3]
```diff
@@ -376,6 +376,33 @@ F\with($value, function($value) {
...
+    return strcmp($value1, $value2);
+});
+
+// Sorts a collection alphabetically, preserving keys
+F\sort($collection, function($value1, $value2, true) {
```
Comment (lstrojny) [inline README.md]: And here

**Comment by lstrojny** [comment]
Can you fix the README and squash the commits? Then we are good to go.

**Comment by colinodell** [comment]
Thanks for the feedback!  Changes have been made and squashed.

**Comment by lstrojny** [comment]
Awesome, thank you!

**Comment by FoxxMD** [comment]
I'm not seeing this the composer package, do you know when it will be updated?

**Comment by lstrojny** [comment]
Will roll a release soon

**Pull request merged**